core/src/main/java/com/sequenceiq/cloudbreak/repository/InstanceMetaDataRepository.java

```diff
@@ -332,4 +333,11 @@ List<InstanceMetadataViewDelegate> findInstanceMetadataViewsByStackIdAndPrivateI
             "WHERE s.id = :stackId " +
             "ORDER BY im.privateId desc")
     Page<Long> findLastPrivateIdForStack(@Param("stackId") Long stackId, Pageable page);
+
+    @Query("SELECT distinct new com.sequenceiq.cloudbreak.dto.SubnetIdWithResourceNameAndCrn(s.name, s.resourceCrn, im.subnetId, s.type) " +
+            "FROM InstanceMetaData im " +
+            "LEFT JOIN im.instanceGroup ig  " +
+            "LEFT JOIN ig.stack s " +
+            "WHERE s.environmentCrn = :environmentCrn")
```

Comment:
wouldn't this return deleted/terminated instances/stacks?

Reply:
good catch, thanks